### PR TITLE
Updated runCommand to reflect builder changes, added searchConfig obj…

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**
…ect.

### CODE REVIEW

File: src/add-search-spec.js [hunk 1]
```diff
@@ -144,9 +144,10 @@ describe('Search Results', () => {
 
 function addSearchSpecToProject(argv, config) {
   if (config &&
-    config.appSettings &&
-    config.appSettings.stache &&
-    config.appSettings.stache.search) {
+      config.appSettings &&
+      config.appSettings.stache &&
+      config.appSettings.stache.searchConfig &&
+      config.appSettings.stache.searchConfig.allowSiteToBeSearched) {
```
Comment (Blackbaud-BobbyEarl) [inline src/add-search-spec.js]: This is a bit of a generic question.  To add searching to a site, is it correct that users were need to add the plugin and add this property to `skyuxconfig.json`?  

If so, I know we could set those defaults in the template they clone, but I'm wondering if we should change the default of `allowSiteToBeSearched` to true?

**Comment by blackbaud-brandonhare** [comment]
I will definitely do an accompanying update to the template to account for the search stuff, yes.

I agree, I think setting the default to true makes the most sense. I think this should be opt out instead of opt in.

File: src/add-search-spec.js (hunk 1 above)
Comment (blackbaud-brandonhare) [reply src/add-search-spec.js]: That's a good idea, but where should that default live? Should I assume it to always be true and then check for the false here?

File: src/remove-search-spec.js [hunk 2]
```diff
@@ -2,21 +2,22 @@
...
-      config.appSettings.stache.searchConfig &&
-      config.appSettings.stache.searchConfig.allowSiteToBeSearched !== false) {
-    try {
-      let filePath = path.join(process.cwd(), 'e2e', 'stache-search.e2e-spec.ts');
-      if (fs.existsSync(filePath)) {
-        fs.unlinkSync(filePath);
-      }
-    } catch (error) {
-      throw new Error('[ERROR]: Unable to remove stache search template from e2e directory.');
+  let doesSearchConfigExist = utils.checkConfig(config, 'allowSiteToBeSearched');
```
Comment (Blackbaud-BobbyEarl) [inline src/remove-search-spec.js]: Any thought on instead having checkConfig be more of a "safe read?"  Something like...
```
if (utils.readConfig(config, 'allowSiteToBeSearch') === false) {
...
}

function readConfig(config, key) {
  if (config &&
  config.appSettings &&
  config.appSettings.stache &&
  config.appSettings.stache.searchConfig) {
    return config.appSettings.stache.searchConfig[key];
  }

  // Not found, "returns" undefined
}
```

File: src/remove-search-json.js [hunk 3]
```diff
@@ -2,22 +2,23 @@
...
-      config.appSettings.stache.searchConfig.allowSiteToBeSearched !== false) {
-    try {
-      let filePath = path.join(process.cwd(), 'src', 'stache', 'search', 'search.json');
-      if (fs.existsSync(filePath)) {
-        fs.unlinkSync(filePath);
-        fs.rmdirSync(filePath.slice(0, -11));
-      }
-    } catch (error) {
-      throw new Error('[ERROR]: Unable to remove stache search directory.');
+  let doesSearchConfigExist = utils.checkConfig(config, 'allowSiteToBeSearched');
```
Comment (Blackbaud-BobbyEarl) [inline src/remove-search-json.js]: Any thought on instead having checkConfig be more of a "safe read?"  Something like...
```
if (utils.readConfig(config, 'allowSiteToBeSearch') === false) {
...
}

function readConfig(config, key) {
  if (config &&
  config.appSettings &&
  config.appSettings.stache &&
  config.appSettings.stache.searchConfig) {
    return config.appSettings.stache.searchConfig[key];
  }

  // Not found, "returns" undefined
}
```

File: src/remove-search-json.js [hunk 4]
```diff
@@ -2,22 +2,23 @@
...
+  if (
+    doesSearchConfigExist &&
+    config.appSettings.stache.searchConfig.allowSiteToBeSearched === false
+  ) { return; }
+  
+  try {
+    let filePath = path.join(process.cwd(), 'src', 'stache', 'search', 'search.json');
```
Comment (Blackbaud-BobbyEarl) [inline src/remove-search-json.js]: I've used this behavior in the past, but somewhat recently have changed to always using resolve + not using multiple parameters.  The resolve just for safety, and the not multiple parameters are fixed in `resolve()` automatically to the OS specific delimiter.

Also, you could probably use a different technique to remove the directory without hard-coding the string length.

```
const filePath = path.resolve(process.cwd(), 'src/stache/search/search.json');
...
fs.rmdirSync(path.dir(filePath));
```

**Comment by blackbaud-brandonhare** [comment]
@Blackbaud-BobbyEarl and @Blackbaud-SteveBrush This is ready for a final review I think. Thanks guys!

File: src/add-search-spec.js [hunk 5]
```diff
@@ -53,13 +54,20 @@ describe('Search Results', () => {
...
+      config.skyux.appSettings.stache &&
+      config.skyux.appSettings.stache.searchConfig
+    );
     let siteName = config.skyux.name;
     if (!siteName) {
       const packageFile = require('../package.json');
       siteName = packageFile.name;
     }
     let url = config.skyux.host.url;
+    let isInternal = doesSearchConfigExist ? config.skyux.appSettings.stache.searchConfig.is_internal : true;
```
Comment (Blackbaud-BobbyEarl) [inline src/add-search-spec.js]: I'm curious why the property uses an underscore instead of camel case?

Comment (blackbaud-brandonhare) [reply src/add-search-spec.js]: So for the properties that are only used by the skyuxconfig, I use camel case. For .net services, the standard is snake case for some reason, so I am using snake case for all properties that get pushed to the backend as well. It is a bit weird, but it helps me keep things that are relevant to my scraped objects separate from config that's only for the SPA and cli commands. Perhaps I should just have them all be camel case for consistency, but that was my initial thought behind it.

File: src/add-search-spec.js [hunk 6]
```diff
@@ -143,19 +152,16 @@ describe('Search Results', () => {
...
-    config.appSettings.stache.search) {
-    try {
-      let filePath = path.join(process.cwd(), 'e2e');
-      if (!fs.existsSync(filePath)) {
-        fs.mkdirSync(filePath);
-      }
-      fs.writeFileSync(path.join(filePath, 'stache-search.e2e-spec.ts'), template);
-    } catch (error) {
-      return errorHandler(new Error('[ERROR]: Unable to add stache search template to e2e directory.'), config);
+  if (utils.readConfig(config, 'allowSiteToBeSearched') === false) { return; }
```
Comment (Blackbaud-SteveBrush) [inline src/add-search-spec.js]: Non-blocking, but these one-line `if` statements through me off.

File: src/publish-search.js [hunk 7]
```diff
@@ -1,12 +1,16 @@
...
 const fs = require('fs-extra');
 const request = require('request');
 const path = require('path');
-const endpoint = process.env.searchEndpoint;
-const token = process.env.token;
-const filePath = path.join(process.cwd(), 'src', 'stache', 'search', 'search.json');
+const audienceId = process.env.audienceId;
+const clientUserName = process.env.clientUserName;
+const clientKey = process.env.clientKey;
+const endpoint = process.env.endpoint;
```
Comment (Blackbaud-SteveBrush) [inline src/publish-search.js]: Non-blocking, but the following format might be easier to modify/read later on.

```
const {
  audienceId,
  clientUserName,
  clientKey,
  endpoint
} = process.env;
```

File: src/publish-search.js [hunk 8]
```diff
@@ -18,42 +22,67 @@ function getSearchData() {
 }
 
 function publishSearch(argv, config) {
-  if (!(config &&
-    config.appSettings &&
-    config.appSettings.stache &&
-    config.appSettings.stache.search)) {
-    return;
-  }
+  if (utils.readConfig(config, 'allowSiteToBeSearched') === false) { return; }
```
Comment (Blackbaud-SteveBrush) [inline src/publish-search.js]: Is this a bad habit you learned from C#? :-P

**Comment by Blackbaud-SteveBrush** [comment]
Aside from some non-blocking style-related comments, this looks good to me. Can you provide a quick checklist I can run through to test this locally?

File: src/publish-search.js (hunk 7 above)
Comment (blackbaud-brandonhare) [reply src/publish-search.js]: Good call! I forget about destructuring sometimes.

**Comment by blackbaud-brandonhare** [comment]
Sure @Blackbaud-SteveBrush!
- Open any stache2 project (I've been using stache2 or stache2-docs);
- Install this package as a dev dependency as "@blackbaud/skyux-builder-stache-search"
- Run the commands using skyux 'command-name'

That's it. The only thing you'll want to do is run the e2e process after you add the spec so you can confirm you get the data and so you have something to manipulate with the other commands. Also, if the e2e step doesn't work with the e2e config, that's due to a bug that I fixed in skyux-builder. I don't think it's been released yet, so just remove omnibar and set auth to false before you run e2e and it'll work fine. You also won't be able to make the publisher call, since you don't have the env variables configured, but I can send you them if you want to test it out and make sure it makes the call properly (the server isn't live yet, so you won't be able to publish anything unless you want to clone down my backend and serve it locally.)

**Comment by blackbaud-brandonhare** [comment]
@Blackbaud-SteveBrush and @Blackbaud-BobbyEarl I've made the changes Bobby suggested. If one of you wouldn't mind giving this a final pass, I think this is ready to go.

File: src/add-search-spec.js [hunk 9]
```diff
@@ -41,25 +46,38 @@ describe('Search Results', () => {
...
-      }
-      return route;
-    });
   });
 
   it('should generate search results', (done) => {
-    let config = JSON.parse(browser.params.skyPagesConfig);
-    let siteName = config.skyux.name;
+    // Comment on why we need these separate configs for the url
+    let config: any = browser.params.skyPagesConfig;
```
Comment (Blackbaud-BrandonJones) [inline src/add-search-spec.js]: If these variables never change, would it be better to use `const` so we can be sure they don't get altered later?

**Pull request merged**